api/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var opentelemetrycollectorlog = logf.Log.WithName("opentelemetrycollector-resource")
+
+func (r *OpenTelemetryCollector) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-opentelemetry-io-v1alpha1-opentelemetrycollector,mutating=true,failurePolicy=fail,groups=opentelemetry.io,resources=opentelemetrycollectors,verbs=create;update,versions=v1alpha1,name=mopentelemetrycollector.kb.io
+
+var _ webhook.Defaulter = &OpenTelemetryCollector{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *OpenTelemetryCollector) Default() {
+	opentelemetrycollectorlog.Info("default", "name", r.Name)
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// +kubebuilder:webhook:verbs=create;update,path=/validate-opentelemetry-io-v1alpha1-opentelemetrycollector,mutating=false,failurePolicy=fail,groups=opentelemetry.io,resources=opentelemetrycollectors,versions=v1alpha1,name=vopentelemetrycollector.kb.io
```

Comment:
According to the above comment, should the `delete` be added to the verbs, given the `ValidateDelete` function is included below? And then either way, the TODO comment should probably be removed - or atleast remove the TODO and explain the current decision.

Reply:
From the manifest it looks like `delete` isn't wanted yet?

Reply:
The manifest is updated based on the tag above. Your point about the existence of `ValidateDelete` is a good one, I'll add `delete` to the list.